app/features/bookmarks/bookmark.service.server.ts

```diff
@@ -4,24 +4,27 @@ import type { Bookmark } from "./bookmark.types";
 
 const apiRequest = createApiRequest(getConfigEntry("API_URL"));
 
-const getAll = async (): Promise<Bookmark[]> => {
-  return await apiRequest.get("/bookmarks");
+const getAll = async () => {
+  let result = await apiRequest.get<Bookmark[]>("/bookmarks");
```

Comment:
Curious on your opinion of passing a generic to `apiRequest.get()` and then letting typescript infer the return type of this method. As opposed to explicitly typing it with `: Promise<Bookmark[]>` like we had.

I don't feel strongly either way. 
- There's a benefit to specifically typing the return type so no one accidentaly changes the return shape.
- But i think there is also a benefit to keeping the code as clean and simple as possible for this demo/start.